samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -353,13 +353,16 @@ class OffsetManager(
         }
       }
 
-      // invoke checkpoint listeners
-      checkpoint.getOffsets.asScala.groupBy { case (ssp, _) => ssp.getSystem }.foreach {
-        case (systemName:String, offsets: Map[SystemStreamPartition, String]) => {
-          // Option is empty if there is no checkpointListener for this systemName
-          checkpointListeners.get(systemName).foreach(_.onCheckpoint(offsets.asJava))
+      // invoke checkpoint listeners only for SSPs that are registered with the OffsetManager
+      val registeredSSPs = systemStreamPartitions.getOrElse(taskName, immutable.Set[SystemStreamPartition]())
```

Comment:
Minor: Can we do the following?
`val registeredSSPs = systemStreamPartitions.getOrElse(taskName, Set())`

Reply:
We need to indicate the type somewhere or the compiler throws up. E.g.
`val registeredSSPs = systemStreamPartitions.getOrElse(taskName, Set[SystemStreamPartition]()`